### ISSUE TITLE
Nullable types in Dictionary values don't seem to be treated as plain Nullable values

### ISSUE THREAD

**Issue by francotiveron** [Zaid-Ajaj/Hawaii#44]
type
```
type Trade() = 
    [<Required>] member val EntryUtc : DateTimeOffset = _default with get, set
    [<Required>] member val EntryGrid : sbyte = _default with get, set
    member val ExitUtc : Nullable<DateTimeOffset> = _default with get, set
    member val ExitGrid : Nullable<sbyte> = _default with get, set
    interface IYatpGraphItem with member this.Utc = this.EntryUtc
```
schema json
```
      "Trade": {
        "required": [
          "EntryGrid",
          "EntryUtc"
        ],
        "type": "object",
        "properties": {
          "EntryUtc": {
            "type": "string",
            "format": "date-time"
          },
          "EntryGrid": {
            "type": "integer",
            "format": "int32"
          },
          "ExitUtc": {
            "type": "string",
            "format": "date-time",
            "nullable": true
          },
          "ExitGrid": {
            "type": "integer",
            "format": "int32",
            "nullable": true
          }
        },
```
Hawaii.NET
```
type Trade =
    { EntryUtc: System.DateTimeOffset
      EntryGrid: int
      ExitUtc: Option<System.DateTimeOffset>
      ExitGrid: Option<int> }
    ///Creates an instance of Trade with all optional fields initialized to None. The required fields are parameters of this function
    static member Create (entryUtc: System.DateTimeOffset, entryGrid: int): Trade =
        { EntryUtc = entryUtc
          EntryGrid = entryGrid
          ExitUtc = None
          ExitGrid = None }

```

**Nullables become Options**, as expected

**INSTEAD**

type
```
type Mdp() = 
    [<Required>] member val Utc : DateTimeOffset = _default with get, set
    [<Required>] member val Volume : int = _default with get, set
    [<Required>] member val Open : float32 = _default with get, set
    [<Required>] member val High : float32 = _default with get, set
    [<Required>] member val Low : float32 = _default with get, set
    [<Required>] member val Close : float32 = _default with get, set
    [<Required>] member val Wave : Dictionary<Episode, Nullable<float32>> = _default with get, set
    [<Required>] member val Indicators : Dictionary<IndicatorId, Nullable<float32>> = _default with get, set
    [<Required>] member val Cards : Dictionary<CardId, CardValue> = _default with get, set
    interface IYatpGraphItem with member this.Utc = this.Utc
```
schema json
```
     "Mdp": {
        "required": [
          "Cards",
          "Close",
          "High",
          "Indicators",
          "Low",
          "Open",
          "Utc",
          "Volume",
          "Wave"
        ],
        "type": "object",
        "properties": {
          "Utc": {
            "type": "string",
            "format": "date-time"
          },
          "Volume": {
            "type": "integer",
            "format": "int32"
          },
          "Open": {
            "type": "number",
            "format": "float"
          },
          "High": {
            "type": "number",
            "format": "float"
          },
          "Low": {
            "type": "number",
            "format": "float"
          },
          "Close": {
            "type": "number",
            "format": "float"
          },
          "Wave": {
            "type": "object",
            "additionalProperties": {
              "type": "number",
              "format": "float",
              "nullable": true
            }
          },
          "Indicators": {
            "type": "object",
            "additionalProperties": {
              "type": "number",
              "format": "float",
              "nullable": true
            }
          },
          "Cards": {
            "type": "object",
            "additionalProperties": {
              "type": "string",
              "nullable": true
            }
          }
        },
```
Note _nullable=true_ are present

Hawaii.NET
```
type Mdp =
    { Utc: System.DateTimeOffset
      Volume: int
      Open: float32
      High: float32
      Low: float32
      Close: float32
      Wave: Map<string, float32>
      Indicators: Map<string, float32>
      Cards: Map<string, string> }
    ///Creates an instance of Mdp with all optional fields initialized to None. The required fields are parameters of this function
    static member Create (utc: System.DateTimeOffset,
                          volume: int,
                          ``open``: float32,
                          high: float32,
                          low: float32,
                          close: float32,
                          wave: Map<string, float32>,
                          indicators: Map<string, float32>,
                          cards: Map<string, string>): Mdp =
        { Utc = utc
          Volume = volume
          Open = ``open``
          High = high
          Low = low
          Close = close
          Wave = wave
          Indicators = indicators
          Cards = cards }
```
Dictionaries becomes Maps, but the value type doesn't respect _nullable=true_ (should be Option<_>)

As a result:

```
 ---> Newtonsoft.Json.JsonSerializationException: Error converting value {null} to type 'System.Single'. Path 'M', line 1, position 171949.
 ---> System.InvalidCastException: Null object cannot be converted to a value type.
```

Note also that if I change manually the generated type from Map<string, t> to Map<string, t option> then it works correctly

**Comment by Zaid-Ajaj** [comment]
Hi @francotiveron this sounds like a reasonable ask 😄 I will have a look when time permits! That said, PRs are always nice 🙏  will be happy to review, merge and publish a package if you can take a stab at it.

**Comment by francotiveron** [comment]
@Zaid-Ajaj I had a quick look at the code. Can you give me some hints, like the source code file where dictionary values are converted? If you can, I could give it a try without searching too long

**Comment by Zaid-Ajaj** [comment]
@francotiveron the issue is probably here https://github.com/Zaid-Ajaj/Hawaii/blob/master/src/Program.fs#L876-L879

**Comment by Zaid-Ajaj** [comment]
Closing as fixed by #45 please re-open if the problem persists 😄
